cmd/maya-apiserver/cstor-operator/cspc/handler.go

```diff
@@ -122,26 +122,41 @@ func (c *Controller) syncCSPC(cspc *apis.CStorPoolCluster) error {
 	//}
 	openebsNameSpace := env.Get(env.OpenEBSNamespace)
 	if openebsNameSpace == "" {
+
+		message := fmt.Sprint("Could not sync CSPC: got empty namespace for openebs from env variable")
+		c.recorder.Event(cspc, apis.EventTypeError, "Getting Namespace", message)
 		return errors.Errorf("Could not sync CSPC {%s}: got empty namespace for openebs from env variable", cspc.Name)
 	}
+
 	pc, err := c.NewPoolConfig(cspc, openebsNameSpace)
 	if err != nil {
+
+		message := fmt.Sprintf("Could not sync CSPC : failed to get pool config: %s", err.Error())
+		c.recorder.Event(cspc, apis.EventTypeError, "Creating Pool Config", message)
 		return errors.Wrapf(err, "Could not sync CSPC {%s}: failed to get pool config", cspc.Name)
 	}
+
 	pendingPoolCount, err := pc.AlgorithmConfig.GetPendingPoolCount()
 	if err != nil {
-		return err
+
+		message := fmt.Sprintf("Could not sync CSPC : failed to get pending pool count: %s", err.Error())
+		c.recorder.Event(cspc, apis.EventTypeError, "Getting Pending Pool(s) ", message)
+		return errors.Wrapf(err, "Could not sync CSPC {%s}: failed to get pending pool count", cspc.Name)
 	}
+
 	if pendingPoolCount > 0 {
 		err = pc.create(pendingPoolCount, cspc)
 		if err != nil {
-			return err
+			message := fmt.Sprintf("Could not create pool(s) for CSPC: %s", err.Error())
+			c.recorder.Event(cspc, apis.EventTypeError, "Pool Create", message)
```

Comment:
Here currently we are getting `error only if we are not able to acquire lease` is it really necessary to keep in front of users about lease errors?

Reply:
I think so. Else how will we know what caused the failure.